flocker/testtools/test/test_flaky.py

```diff
@@ -42,63 +82,138 @@ def test_successful_flaky_test(self):
 
         # We use 'unittest' here to avoid accidentally depending on Twisted
         # TestCase features, thus increasing complexity.
-        class SomeTest(unittest.TestCase):
+        class SomeTest(AsyncTestCase):
 
-            @flaky('FLOC-XXXX')
+            @flaky(u'FLOC-XXXX')
             def test_something(self):
                 pass
 
         test = SomeTest('test_something')
-        self.assertEqual({
-            'errors': 0,
-            'failures': 0,
-            'skipped': 0,
-            'expectedFailures': 0,
-            'unexpectedSuccesses': 0,
-            'testsRun': 1,
-        }, get_results(test))
-
-    def test_failed_flaky_test(self):
+        self.assertThat(run_test(test), has_results(tests_run=Equals(1)))
+
+    def test_always_erroring_flaky_test(self):
+        """
+        A flaky test always errors out is recorded as erroring.
+        """
+
+        executions = repeat(lambda: throw(ValueError('failure')))
+
+        class SomeTest(AsyncTestCase):
+
+            @flaky(u'FLOC-XXXX')
+            def test_something(self):
+                next(executions)()
+
+        test = SomeTest('test_something')
+        self.assertThat(
+            run_test(test), has_results(
+                errors=HasLength(1),
+                tests_run=Equals(1),
+            )
+        )
+
+    def test_intermittent_flaky_test(self):
         """
-        As of FLOC-3414, the @flaky decorator doesn't actually treat failed
-        tests in special in any way - it just acts as a structured comment.
+        A @flaky test that fails sometimes and succeeds other times counts as a
+        pass, as long as it passes more than the given min_passes threshold.
         """
+        # XXX: Do we have a 'shuffled' strategy?
```

Comment:
I think https://github.com/DRMacIver/hypothesis/blob/49d860822639c2426a9bcd9ebf55c0b6e0ee472f/src/hypothesis/strategies.py#L706 might be what you are looking for?